### ISSUE TITLE
How to load only a part of pretrained model weights?

### ISSUE THREAD

**Issue by JulioZhao97** [ultralytics/ultralytics#18625]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

In only want to load a part of pretrained model weights, such as only loading the weights of first 3 backbone stages out of 4. How can I do that?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JulioZhao97, thank you for your interest in Ultralytics 🚀! We recommend checking the [Docs](https://docs.ultralytics.com) for guidance and common usage examples, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) workflows.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us investigate it.

Regarding your question ❓ about loading specific parts of pretrained model weights, we encourage providing additional details such as the model type, the specific layers or components you wish to load, and any code or workflow attempts you’ve already tried. This will assist both us and the community in providing the most accurate guidance!

In the meantime, you can join the vibrant Ultralytics community for further support:
- For real-time chat, join us on [Discord](https://discord.com/invite/ultralytics) 🎧
- Engage in detailed discussions on our [Discourse](https://community.ultralytics.com)
- Explore insights or share knowledge on the [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Ensure you're using the latest version to access all the newest features and fixes. Update your `ultralytics` package in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

Here are some verified YOLO environments you can explore for quick experimentation:
- **Notebooks** with free GPU:
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM ([Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/))
- **Amazon** Deep Learning AMI ([Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/))
- **Docker Image** ([Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

Check our CI test status for verified compatibility with various environments and use cases:
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

An Ultralytics engineer will review your question soon. Thank you for your patience! 😊

**Comment by Y-T-G** [comment]
```python
pretrained  = YOLO("pretrained.pt")
pretrained.model.model = pretrained.model.model[:4]  # Keep only first 4 layers
pretrained.save("pretrained_4.pt")
```

Then just load `model.load("pretrained_4.pt")`

**Comment by JulioZhao97** [comment]
@Y-T-G Thanks!

**Comment by JulioZhao97** [comment]
It turns out the above method do not work at all, after model calls init_process_group(), the model weights is re-initialized.

**Comment by Y-T-G** [comment]
If you saved it as a file, it will work. Otherwise, it won't.
